Upstream-Status: Pending

Check marco OPENSSL_NO_DES before codes require openssl des header files and
libraries.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
diff --git a/src/racoon/algorithm.c b/src/racoon/algorithm.c
index 3fd50f6..7842cc6 100644
--- a/src/racoon/algorithm.c
+++ b/src/racoon/algorithm.c
@@ -103,9 +103,11 @@ static struct hmac_algorithm oakley_hmacdef[] = {
 };
 
 static struct enc_algorithm oakley_encdef[] = {
+#ifndef OPENSSL_NO_DES
 { "des",	algtype_des,		OAKLEY_ATTR_ENC_ALG_DES,	8,
 		eay_des_encrypt,	eay_des_decrypt,
 		eay_des_weakkey,	eay_des_keylen, },
+#endif
 #ifdef HAVE_OPENSSL_IDEA_H
 { "idea",	algtype_idea,		OAKLEY_ATTR_ENC_ALG_IDEA,	8,
 		eay_idea_encrypt,	eay_idea_decrypt,
@@ -119,9 +121,11 @@ static struct enc_algorithm oakley_encdef[] = {
 		eay_rc5_encrypt,	eay_rc5_decrypt,
 		eay_rc5_weakkey,	eay_rc5_keylen, },
 #endif
+#ifndef OPENSSL_NO_DES
 { "3des",	algtype_3des,		OAKLEY_ATTR_ENC_ALG_3DES,	8,
 		eay_3des_encrypt,	eay_3des_decrypt,
 		eay_3des_weakkey,	eay_3des_keylen, },
+#endif
 { "cast",	algtype_cast128,	OAKLEY_ATTR_ENC_ALG_CAST,	8,
 		eay_cast_encrypt,	eay_cast_decrypt,
 		eay_cast_weakkey,	eay_cast_keylen, },
@@ -136,6 +140,7 @@ static struct enc_algorithm oakley_encdef[] = {
 };
 
 static struct enc_algorithm ipsec_encdef[] = {
+#ifndef OPENSSL_NO_DES
 { "des-iv64",	algtype_des_iv64,	IPSECDOI_ESP_DES_IV64,		8,
 		NULL,			NULL,
 		NULL,			eay_des_keylen, },
@@ -145,6 +150,7 @@ static struct enc_algorithm ipsec_encdef[] = {
 { "3des",	algtype_3des,		IPSECDOI_ESP_3DES,		8,
 		NULL,			NULL,
 		NULL,			eay_3des_keylen, },
+#endif
 #ifdef HAVE_OPENSSL_RC5_H
 { "rc5",	algtype_rc5,		IPSECDOI_ESP_RC5,		8,
 		NULL,			NULL,
@@ -156,9 +162,11 @@ static struct enc_algorithm ipsec_encdef[] = {
 { "blowfish",	algtype_blowfish,	IPSECDOI_ESP_BLOWFISH,		8,
 		NULL,			NULL,
 		NULL,			eay_bf_keylen, },
+#ifndef OPENSSL_NO_DES
 { "des-iv32",	algtype_des_iv32,	IPSECDOI_ESP_DES_IV32,		8,
 		NULL,			NULL,
 		NULL,			eay_des_keylen, },
+#endif
 { "null",	algtype_null_enc,	IPSECDOI_ESP_NULL,		8,
 		NULL,			NULL,
 		NULL,			eay_null_keylen, },
diff --git a/src/racoon/crypto_openssl.c b/src/racoon/crypto_openssl.c
index 55b076a..f98495b 100644
--- a/src/racoon/crypto_openssl.c
+++ b/src/racoon/crypto_openssl.c
@@ -58,7 +58,11 @@
 #include <openssl/md5.h>
 #include <openssl/sha.h>
 #include <openssl/hmac.h>
+
+#ifndef OPENSSL_NO_DES
 #include <openssl/des.h>
+#endif
+
 #include <openssl/crypto.h>
 #ifdef HAVE_OPENSSL_ENGINE_H
 #include <openssl/engine.h>
@@ -1195,6 +1199,7 @@ evp_keylen(int len, const EVP_CIPHER *e)
 	return EVP_CIPHER_key_length(e) << 3;
 }
 
+#ifndef OPENSSL_NO_DES
 /*
  * DES-CBC
  */
@@ -1229,6 +1234,7 @@ eay_des_keylen(len)
 {
 	return evp_keylen(len, EVP_des_cbc());
 }
+#endif
 
 #ifdef HAVE_OPENSSL_IDEA_H
 /*
@@ -1393,6 +1399,7 @@ eay_rc5_keylen(len)
 }
 #endif
 
+#ifndef OPENSSL_NO_DES
 /*
  * 3DES-CBC
  */
@@ -1427,6 +1434,8 @@ eay_3des_weakkey(key)
 	    des_is_weak_key((void *)(key->v + 16)));
 #endif
 }
+#endif
+
 
 int
 eay_3des_keylen(len)
diff --git a/src/racoon/eaytest.c b/src/racoon/eaytest.c
index 1474bdc..54d669a 100644
--- a/src/racoon/eaytest.c
+++ b/src/racoon/eaytest.c
@@ -663,6 +663,7 @@ ciphertest(ac, av)
 	key.v = str2val("f59bd70f 81b9b9cc 2a32c7fd 229a4b37", 16, &key.l);
 	iv0.v = str2val("26b68c90 9467b4ab 7ec29fa0 0b696b55", 16, &iv0.l);
 
+#ifndef OPENSSL_NO_DES
 	if (ciphertest_1 ("DES", 
 			  &data, 8, 
 			  &key, 8, 
@@ -676,6 +677,7 @@ ciphertest(ac, av)
 			  &iv0, 8,
 			  eay_3des_encrypt, eay_3des_decrypt) < 0)
 	  return -1;
+#endif
 	
 	if (ciphertest_1 ("AES",
 			  &data, 16,
